### PR TITLE
feat: add a helper class to ease live preview implementation

### PR DESCRIPTION
**Pull request by faraazb**
Fixes: https://contentstack.atlassian.net/browse/VC-212

- Add a helper class - `ContentstackGatsby` with a `get()` method to fetch live preview data
- Add a hook - `useContentstackImage` to pass a Contentstack image URL and get Gatsby image data for use with [`GatsbyImage`](https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-plugin-image#gatsbyimage) component

### CODE REVIEW

File: src/live-preview/storage-helper.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    storage;
+    name;
+
+    /**
+     * @param {string} name 
+     * @param {Storage.prototype} storage 
+     */
+    constructor(storage, name) {
+        if (!storage?.__proto__ === Storage.prototype) {
+            throw new Error("storage should implment Storage")
```
Comment (abhishek305) [inline src/live-preview/storage-helper.js:15]: could you verify this statement looks a bit off

**Review by abhishek305 — CHANGES_REQUESTED** [review]

File: src/live-preview/storage-helper.js (hunk 1 above)
Comment (faraazb) [reply src/live-preview/storage-helper.js:15]: Ensures that the storage parameter is implementing the [Web Storage API](https://developer.mozilla.org/en-US/docs/Web/API/Storage). So the class can be used with local or session or any Storage object.

File: src/live-preview/index.js [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+export class ContentstackGatsby {
+    config;
+    stackSdk;
+    contentTypes;
+    referenceFields;
+    referenceFieldPaths;
+
+    constructor(config) {
+        this.config = config;
```
[the remark this answers is not in the document]
Comment (faraazb) [reply src/live-preview/index.js:17]: No, this config is the config provided by the user while initializing `ContentstackGatsby`

File: src/live-preview/index.js [hunk 3]
```diff
@@ -0,0 +1,248 @@
+import contentstack from "contentstack"
+import { jsonToHTML } from "@contentstack/utils"
+import isEmpty from "lodash.isempty";
+import { Storage } from "./storage-helper";
+
+// max depth for nested references
+const MAX_DEPTH_ALLOWED = 5;
```
[the remark this answers is not in the document]
Comment (faraazb) [reply src/live-preview/index.js:7]: The CMA API resolves nested references upto this depth (5). It returns an error response (4XX) if we ask for a deeper reference.

**Review by abhishek305 — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -27,8 +27,10 @@
   "license": "MIT",
```
[the remark this answers is not in the document]
Comment (faraazb) [reply package.json:27]: Bumped to 5.1.0

**Review by surajcontentstack — APPROVED** [review]

**Pull request merged**